src/NuGet.Clients/VsExtension/NuGetPackage.cs

```diff
@@ -390,10 +390,10 @@ private IEnumerable<NuGet.Credentials.ICredentialProvider> GetCredentialProvider
                 VisualStudioAccountProvider.FactoryMethod,
                 errorDelegate);
 
-            var vstsProvider = importer.GetProvider();
-            if (vstsProvider != null)
+            var importedProviders = importer.GetProviders();
+            if (importedProviders.Count != 0)
             {
-                credentialProviders.Add(vstsProvider);
+                credentialProviders.AddRange(importedProviders);
             }
 
             var webProxy = (IVsWebProxy)GetService(typeof(SVsWebProxy));
```

Comment:
Looking at the [CredentialService](https://github.com/NuGet/NuGet.Client/blob/3b239747175ca9ca09bc7be3e46c46c6f856454e/src/NuGet.Clients/NuGet.Credentials/CredentialService.cs#L95), it just does a foreach over the elements in the order they were added to the `List`.

I don't know whether the VSTS provider relies on ordering of the other providers, but given that has already been tested using the current ordering, the VSTS provider should just keep working. @pspill may know more about the importance of ordering these providers in this method.

If ordering is indeed important, I'd like to learn why (+ this may need to be documented in the code here). We could then consider ordering the imported providers (maybe using MEF Metadata on the ImportMany, depending on ordering criteria).